### PR TITLE
fix: A bug fix - avm vmss bootdiagnosticenabled

### LINKED ISSUE #3713: [AVM Module Issue]: Boot Diagnostics has wrong parameter value
### Check for previous/existing GitHub issues

- [x] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### Module Name

avm/res/compute/virtual-machine-scale-set

### (Optional) Module Version

0.4.0

### Description

Unable to enable Boot Diagnostics, because parameter is configured wrong. It needs boolen, but requesting a string. After looking into it more deeply, the Microsoft best practice is to not include a storage account, but that can be optional. The  [learn docs](https://learn.microsoft.com/en-us/azure/virtual-machines/boot-diagnostics) state that all you need is the enabled `true` or `false`

```
bootDiagnostics: {
          enabled: !empty(bootDiagnosticStorageAccountName)
          storageUri: !empty(bootDiagnosticStorageAccountName)
            ? 'https://${bootDiagnosticStorageAccountName}${bootDiagnosticStorageAccountUri}'
            : null
}
```

Feedback:
After looking into this for VMs and VMSS, I noticed this has the same issue. `bootDiagnosticEnabled ` needs to be a param to allow the users to decide if they should use the Microsoft best practice managed storage or the optional secure standalone storage account. 

Suggested fix:

```
@description('Optional. Enable boot diagnostics to use default managed or secure storage. Defaults to false.')
param bootDiagnosticEnabled bool = false

@description('Optional. The name of the boot diagnostic storage account. Provide this if you want to use your own storage account for security reasons instead of the recommended Microsoft Managed Storage Account.')
param bootDiagnosticStorageAccountName string?

@description('Optional. The URI of the boot diagnostic storage account.')
param bootDiagnosticStorageAccountUri string?

bootDiagnostics: {
  enabled: bootDiagnosticEnabled
  storageUri: !empty(bootDiagnosticStorageAccountName) ? 'https://${bootDiagnosticStorageAccountName}${bootDiagnosticStorageAccountUri}' : null
}

```

Happy to create a PR if needed. 
### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by FallenHoot**
## Description

Unable to enable Boot Diagnostics, because parameter is configured wrong. It needs boolen, but requesting a string. After looking into it more deeply, the Microsoft best practice is to not include a storage account, but that can be optional. The [learn docs](https://learn.microsoft.com/en-us/azure/virtual-machines/boot-diagnostics) state that all you need is the enabled true or false

Fixes #3713

## Pipeline Reference

[avm.res.compute.virtual-machine-scale-set](https://github.com/FallenHoot/bicep-registry-modules/actions/workflows/avm.res.compute.virtual-machine-scale-set.yml)

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [x] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [x] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [X] I have run `Set-AVMModule` locally to generate the supporting module files.
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/compute/virtual-machine/main.bicep
Comment (FallenHoot) [reply avm/res/compute/virtual-machine/main.bicep]: That is correct. I noticed this as well and thought I squashed this commit, but that wasn't the case. I have now restored the virtual-machine files.

**Comment by rahalan** [comment]
@FallenHoot can you please provide a successful run for the workflow. Thanks.

**Comment by rahalan** [comment]
@FallenHoot please update the version file (bump the minor version by 1). Thanks.

**Comment by FallenHoot** [comment]
> @FallenHoot can you please provide a successful run for the workflow. Thanks.

image

**Comment by rahalan** [comment]
@FallenHoot thanks. Actually I am looking for a link to the successful pipeline run (can be the badge). The links should go to something like this: 
image

**Comment by jtracey93** [comment]
Hey @FallenHoot ,

Firstly, thanks for your work on this PR!

We have made some changes to the AVM CI, detailed below, which means we need you to update your fork to pull in these latest changes and re-run your tests to show they still are passing prior to approving and merging this PR, as we don't and it fails once merged the publishing of your module will fail and will be blocked going forward until the test pass again via additional PRs.

### Changes to CI That Have Been Made That You Need To Take Action On

- https://github.com/Azure/bicep-registry-modules/pull/3745
    - Adds WAF Security PSRule Checks using this custom baseline of rules ([`/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/cb-waf-security.Rule.yaml`](https://github.com/Azure/bicep-registry-modules/blob/main/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/cb-waf-security.Rule.yaml))
    - **Action Required By You:**
        - Ingest/pull-in these changes into your PR branch and re-run your tests to ensure they pass
            - If any issues with the rules and you think it should apply, please ping/reach out to the core team

Any questions reach out to the AVM Core Team by tagging us in your PR here or internally via Teams

Thanks

Jack (AVM Core Team)

**Comment by FallenHoot** [comment]
I am running into this error - WriteErrorException: Failed to download telemetry CSV file from [https://aka.ms/avm/index/bicep/res/csv] due to [The SSL connection could not be established, see inner exception.].
  at <ScriptBlock>, /home/runner/work/bicep-registry-modules/bicep-registry-modules/avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1:906

**Comment by FallenHoot** [comment]
Badge is now green

[avm.res.compute.virtual-machine-scale-set](https://github.com/FallenHoot/bicep-registry-modules/actions/workflows/avm.res.compute.virtual-machine-scale-set.yml)

Please do accept and merge @rahalan

File: avm/res/db-for-my-sql/flexible-server/tests/e2e/max/dependencies3.bicep
Comment (arnoldna) [inline avm/res/db-for-my-sql/flexible-server/tests/e2e/max/dependencies3.bicep:1]: I don't think you meant to include this file in your pull request.

File: avm/res/db-for-postgre-sql/flexible-server/README.md
Comment (arnoldna) [inline avm/res/db-for-postgre-sql/flexible-server/README.md:1]: I don't think you meant to include this file in your pull request.

Comment (FallenHoot) [reply avm/res/db-for-postgre-sql/flexible-server/README.md:1]: Nope, I will remove it. Thank you

**Review by rahalan — APPROVED** [review]

**Pull request merged**
